pkg/ring/ring.go

```diff
@@ -326,29 +339,51 @@ func (r *Ring) GetAll(op Operation) (ReplicationSet, error) {
 		return ReplicationSet{}, ErrEmptyRing
 	}
 
+	maxUnavailableZones := len(r.ringZones) / 2
```

Comment:
It was changed to `max(0, len(zones) - maxUnavailable -1)`. Without the minus 1 for example for RF=3 and `len(zones)` = 3 then it would result in 2. Similarly for RF = 5 and `len(zones)` = 5 that would result in 3.